extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -146,12 +146,9 @@ private List<InputRow> parseMetric(Metric metric, Map<String, Object> resourceAt
       }
       // TODO Support HISTOGRAM and SUMMARY metrics
       case HISTOGRAM:
-      case SUMMARY: {
-        inputRows = Collections.emptyList();
-        break;
-      }
+      case SUMMARY:
       default:
-        throw new IllegalStateException("Unexpected value: " + metric.getDataCase());
+        inputRows = Collections.emptyList();
```

Comment:
This is probably OK as a stopgap, but since the contract of `InputEntityReader` is to return an `Iterator` I think we can do better, but it would require a larger refactoring.

Instead of eagerly reading all the metrics in the protobuf into a `List` and then returning `list.iterator()` we should return an `Iterator` that lazily converts the metric from the protobuf into an `InputRow`.  This way we can throw a `ParseException` from `next()` if the data point is invalid.  Hopefully this will only skip the individual row and not the entire batch.